pkg/api/types.go

```diff
@@ -1456,3 +1456,94 @@ func AddToNodeAddresses(addresses *[]NodeAddress, addAddresses ...NodeAddress) {
 		}
 	}
 }
+
+// AutoScaler monitors other resources that are resizeable and adjusts them according to configuration.
+type AutoScaler struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the auto-scaler targets and thresholds.
+	Spec AutoScalerSpec `json:"spec,omitempty"`
+
+	// Status defines the actions the auto-scaler has taken.
+	Status AutoScalerStatus `json:"status,omitempty"`
+}
+
+// AutoScalerStatus provides the status of an auto-scaler.
+type AutoScalerStatus struct {
+	// TODO: open for discussion on what meaningful information can be reported in the status
+	// The status may return the replica count here but we may want more information
+	// such as if the count reflects a threshold being passed.
+}
+
+// AutoScalerSpec defines the auto-scaler targets and thresholds.
+type AutoScalerSpec struct {
+	// Thresholds holds a collection of AutoScaleThresholds that drive the auto-scaler.
+	Thresholds []AutoScaleThreshold `json:"thresholds,omitempty"`
+
+	// Enabled turns auto scaling on or off.
+	Enabled bool `json:"enabled,omitempty"`
+
+	// MaxAutoScaleCount defines the max replicas that the auto-scaler can use.  This value must be
+	// >= MinAutoScaleCount.
+	MaxAutoScaleCount int `json:"maxAutoScaleCount,omitempty"`
+
+	// MinAutoScaleCount defines the minimum number replicas that the auto-scaler can reduce to,
+	// 0 means that the application is allowed to idle.
+	MinAutoScaleCount int `json:"minAutoScaleCount,omitempty"`
+
+	// TargetSelector provides the resizeable target(s).  Right now this is a ReplicationController
+	// in the future it could be a job or any resource that implements resize.  If multiple targets
+	// are resolved by the selector the auto-scaler will resize the largest one.
+	TargetSelector map[string]string `json:"targetSelector,omitempty"`
+
+	// MonitorSelector defines a set of capacity that the auto-scaler is monitoring
+	// (replication controllers).  Monitored objects are used by thresholds to examine
+	// statistics.  Example: get statistic X for object Y to see if threshold is passed.
+	MonitorSelector map[string]string `json:"monitorSelector,omitempty"`
```

Comment:
Put that in the comment :)


> On Mar 16, 2015, at 8:45 AM, Paul <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > +	// >= MinAutoScaleCount.
> > +	MaxAutoScaleCount int `json:"maxAutoScaleCount,omitempty"`
> > +
> > +	// MinAutoScaleCount defines the minimum number replicas that the auto-scaler can reduce to,
> > +	// 0 means that the application is allowed to idle.
> > +	MinAutoScaleCount int `json:"minAutoScaleCount,omitempty"`
> > +
> > +	// TargetSelector provides the resizeable target(s).  Right now this is a ReplicationController
> > +	// in the future it could be a job or any resource that implements resize.  If multiple targets
> > +	// are resolved by the selector the auto-scaler will resize the largest one.
> > +	TargetSelector map[string]string `json:"targetSelector,omitempty"`
> > +
> > +	// MonitorSelector defines a set of capacity that the auto-scaler is monitoring
> > +	// (replication controllers).  Monitored objects are used by thresholds to examine
> > +	// statistics.  Example: get statistic X for object Y to see if threshold is passed.
> > +	MonitorSelector map[string]string `json:"monitorSelector,omitempty"`
> The monitor selector provides the set of objects that they auto-scaler is responsible for monitoring. The thresholds may use these objects to retrieve statistics from the store. In the example object Y is an object returned by the monitor selector.
> 
> —
> Reply to this email directly or view it on GitHub.
>